test/init.js

```diff
@@ -43,9 +43,14 @@ global.getDBConfig = function(useUrl) {
   };
   return settings;
 };
+
+var db;
 global.getDataSource = global.getSchema = function(useUrl) {
+  // Return cached data source if possible to avoid too many client error
+  // due to multiple instances of connection pools
+  if (!useUrl && db) return db;
```

Comment:
I tried to clear the models attached to db in previous tests
```
db.connector._models = {}
```
But it breaks some test like https://github.com/strongloop/loopback-datasource-juggler/blob/master/test/basic-querying.test.js#L60
which runs getSchema() to get db again for `unknownid` generator. Will fix it then.